presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -232,16 +258,22 @@ public boolean isDecomposable(FunctionRegistry functionRegistry)
         private final FunctionCall call;
         private final Signature signature;
         private final Optional<Symbol> mask;
+        private final List<Symbol> orderBy;
+        private final List<SortOrder> ordering;
 
         @JsonCreator
         public Aggregation(
                 @JsonProperty("call") FunctionCall call,
                 @JsonProperty("signature") Signature signature,
-                @JsonProperty("mask") Optional<Symbol> mask)
+                @JsonProperty("mask") Optional<Symbol> mask,
+                @JsonProperty("orderBy") List<Symbol> orderBy,
+                @JsonProperty("ordering") List<SortOrder> ordering)
         {
             this.call = call;
             this.signature = signature;
             this.mask = mask;
+            this.orderBy = orderBy;
+            this.ordering = ordering;
```

Comment:
> It's not validated because others are not validated. I figured I'd keep them looking the same. Should I validate them all?

I fail to see reason for lack of validation, so as usual: validate what you add. Anything else is a separate commit.

> I can change AST to use empty list to mean no order 

i'm not saying you should do, just that you should think about that. @kokosing? you know the conventions better than i do